src/spawn_sync.cc

```diff
@@ -722,45 +743,59 @@ Local<Array> SyncProcessRunner::BuildOutputArray() {
 }
 
 
-int SyncProcessRunner::ParseOptions(Local<Value> js_value) {
+v8::Maybe<int> SyncProcessRunner::ParseOptions(Local<Value> js_value) {
   HandleScope scope(env()->isolate());
   int r;
 
   if (!js_value->IsObject())
-    return UV_EINVAL;
+    return v8::Just<int>(UV_EINVAL);
 
   Local<Context> context = env()->context();
   Local<Object> js_options = js_value.As<Object>();
 
   Local<Value> js_file =
       js_options->Get(context, env()->file_string()).ToLocalChecked();
```

Comment:
Now, I know that you are a huuuuge fan of PR scope creep, but ... feel free to take care of other such places in the functions you're already touching as well, if you like?